tree/arbo/tree.go

```diff
@@ -138,22 +139,27 @@ func NewTreeWithTx(wTx db.WriteTx, cfg Config) (*Tree, error) {
 	if cfg.ThresholdNLeafs == 0 {
 		cfg.ThresholdNLeafs = DefaultThresholdNLeafs
 	}
-	t := Tree{db: cfg.Database, maxLevels: cfg.MaxLevels,
-		thresholdNLeafs: cfg.ThresholdNLeafs, hashFunction: cfg.HashFunction}
+
+	t := Tree{
+		db:              cfg.Database,
+		maxLevels:       cfg.MaxLevels,
+		thresholdNLeafs: cfg.ThresholdNLeafs,
+		hashFunction:    cfg.HashFunction,
+	}
+
 	t.emptyHash = make([]byte, t.hashFunction.Len()) // empty
+	var err error
+	t.emptyNode, _, err = t.newIntermediate(t.emptyHash, t.emptyHash)
```

Comment:
Why the empty node is required?

Reply:
The empty node is used to compare in the up() function if both childs are empty. Precomputing and comparing is faster.